django/thunderstore/community/api/experimental/views.py

```diff
@@ -0,0 +1,32 @@
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+from thunderstore.community.api.experimental.serializers import (
+    CommunitySerializer,
+    PackageCategorySerializer,
+)
+from thunderstore.community.models import Community
+
+
+class CommunitiesExperimentalApiView(APIView):
+    def get(self, request, format=None):
+        communities = CommunitySerializer(Community.objects.listed(), many=True)
+        return Response(
+            {
+                "communities": communities.data,
+            },
+        )
```

Comment:
This could have pagination but it might be worth adding an index, sorting by identifier and using pages? Using pages could mean communities created in-between requests might be a non-issue and I don't think we really have a cursor we can use without exposing the `id`.

Reply:
Cursor should be fine even if it includes the `id` in my opinion, it's not a secret or anything.

If you use a `ListAPIView`, it's relatively easy to add a cursor paginator. This is what we have for the mod package list atm:
```python
class CustomCursorPagination(CursorPagination):
    ordering = "-date_created"
    page_size = 5

class PackageListApiView(ManualCacheMixin, ListAPIView):
    pagination_class = CustomCursorPagination
    ...
```

Reply:
> Cursor should be fine even if it includes the `id` in my opinion, it's not a secret or anything

If we do use the ID we should probably document somewhere our policy for exposing IDs. In some other places, we've intentionally avoided exposing IDs, so this would seem a bit inconsistent without explicit guidelines.

Reply:
The cursor does not expose the ID though. It's encoded yes but it's not a "public interface", since each page gives you a link to the previous page.

Reply:
> The cursor does not expose the ID though. It's encoded yes but it's not a "public interface"

What do you mean?

Reply:
For example, this is the cursor for the experimental API:

https://thunderstore.io/api/experimental/package/?cursor=cD0yMDIxLTAzLTIwKzE1JTNBMjUlM0EwOC4zNDczMDAlMkIwMCUzQTAw

That decodes into
```
p=2021-03-20+15%3A25%3A08.347300%2B00%3A00
```
so looks like the cursor doesn't even expose ID in thef irst place, since date sorting is used here. Even if ID was exposed in the cursor, it would be encoded, and the implication would be that you should not encode/decode cursors yourself but instead use the ones provided by the API

Reply:
I was mainly thinking about user IDs. I thought we were using the original user ID when creating ghost users, but I misremembered and this was only thought about for the `models.SET()` solution (https://github.com/thunderstore-io/Thunderstore/pull/196#discussion_r563905869), not the solution that ended up being used.

I think the comment that made me think we were avoiding showing IDs intentionally was https://github.com/thunderstore-io/Thunderstore/pull/196#discussion_r563908940 but it doesn't sound like this is the case.

I just realised `Community` uses `TimestampMixin` so we actually *do* have a created at timestamp we can use instead of `id` anyway, which I think makes more sense.